app/javascript/src/components/Dashboard/Notes/Pane/Form.jsx

```diff
@@ -1,23 +1,35 @@
 import React, { useState } from "react";
 
 import { Formik, Form } from "formik";
+import { Check } from "neetoicons";
 import { Button, Pane } from "neetoui";
-import { Input, Textarea } from "neetoui/formik";
+import { Input, Select } from "neetoui/formik";
 
-import notesApi from "apis/notes";
+import { CONTACTS, TAGS } from "components/Dashboard/constants";
+import { buildSelectOption } from "utils/index";
 
 import { NOTES_FORM_VALIDATION_SCHEMA } from "../constants";
 
-const NoteForm = ({ onClose, note, isEdit }) => {
+const tags = TAGS.map(buildSelectOption);
+const contacts = CONTACTS.map(contact =>
+  buildSelectOption(`${contact.firstName} ${contact.lastName}`)
+);
+
+const parseValues = values => {
+  const assignedContact = values.assignedContact.label;
+  const tags = values.tags.map(tag => tag.label);
+
+  return { ...values, tags, assignedContact };
+};
+
+const NoteForm = ({ onClose, note, isEdit, handleNote }) => {
   const [submitted, setSubmitted] = useState(false);
 
-  const handleSubmit = async values => {
+  const handleSubmit = values => {
     try {
-      if (isEdit) {
-        await notesApi.update(note.id, values);
-      } else {
-        await notesApi.create(values);
-      }
+      isEdit
+        ? handleNote(note.id, parseValues(values))
+        : handleNote(parseValues(values));
```

Comment:
This is not right. The `handleNote` function is ambiguous right now. How many parameters does it accept? Does it accept positional or named arguments?

This piece of code suggests that the `handleNote` function has two different implementations, which shouldn't happen.

Also the `handleNote` function name doesn't convey what exactly it's handling? Like how exactly does it handle the `note`?

Reply:
Ok, I'll modify those lines, thanks for pointing it out.😀